src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi

```diff
@@ -144,8 +144,14 @@ cdef class SSLChannelCredentials(ChannelCredentials):
       return grpc_ssl_credentials_create(
           c_pem_root_certificates, NULL, NULL, NULL)
     else:
-      c_pem_key_certificate_pair.private_key = self._private_key
-      c_pem_key_certificate_pair.certificate_chain = self._certificate_chain
+      if self._private_key:
```

Comment:
we should add a test that would've detected this bug?

Reply:
The real problem is we are hiding Cython exceptions, not just here. We should revamp Cython code to surface exceptions.